src/slider/slider-labels.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import styles from './styles.css.js';
+import { useMobile } from '../internal/hooks/use-mobile/index.js';
+import customCssProps from '../internal/generated/custom-css-properties/index.js';
+
+export interface SliderLabelProps {
+  min: number;
+  max: number;
+  referenceValues?: ReadonlyArray<number>;
+  valueFormatter?: (value: number) => string;
+  labelsId: string;
+}
+export default function SliderLabels({ min, max, referenceValues, valueFormatter, labelsId }: SliderLabelProps) {
+  const isMobile = useMobile();
+  const LABEL_THRESHOLD = isMobile ? 5 : 10;
+
+  const getVisibleReferenceValues = () => {
+    if (!referenceValues || referenceValues.length === 0) {
+      return [];
+    }
+
+    const values: Array<number> = [];
+    let lastValue = 0;
+    for (let i = 0; i <= referenceValues.length; i++) {
```

Comment:
Do we expect the array to be sorted?

Reply:
No, it shouldn't matter.

Reply:
why not?

Reply:
Because each value is assigned a column in the grid layout, and that's what determines the order. Added a test for that [here](https://github.com/cloudscape-design/components/pull/2056/files#diff-acb4f1937faef0d28137ce656c4de3e363d2215585461f1fc671869bb3f04367R115).